baseplate/context/kombu.py

```diff
@@ -0,0 +1,71 @@
+from __future__ import absolute_import
+from __future__ import division
+from __future__ import print_function
+from __future__ import unicode_literals
+
+from kombu.pools import Producers
+
+from baseplate.context import ContextFactory
+
+
+class KombuProducerContextFactory(ContextFactory):
+    """KombuProducer context factory.
+
+    This factory will attach a proxy object which acts like a
+    :py:class:`kombu.Producer` to an attribute on the :term:`context object`.
+    The ``publish`` method will automatically record diagnostic information.
+
+    :param kombu.connection.Connection connection: A configured connection
+        object.
+    :param kombu.Exchange exchange: A configured exchange object
+    :param int max_connections: The maximum number of connections.
+
+    """
+    def __init__(self, connection, exchange, max_connections=None):
+        self.connection = connection
+        self.exchange = exchange
+        self.producers = Producers(limit=max_connections)
+
+    def make_object_for_context(self, name, span):
+        return KombuProducer(
+            name, span, self.connection, self.exchange, self.producers)
+
+
+class KombuProducer(object):
+    def __init__(self, name, span, connection, exchange, producers):
+        self.name = name
+        self.span = span
+        self.connection = connection
+        self.exchange = exchange
+        self.producers = producers
+
+    def publish(self, body, routing_key=None, **kwargs):
+        """Publish a message to the routing_key.
+
+        :param str body: The message body.
+        :param str routing_key: The AMQP routing key to publish to. If there
```

Comment:
One of the reasons Kombu was chosen over alternatives is that it supports [more than just AMQP](http://docs.celeryproject.org/projects/kombu/en/latest/userguide/connections.html#transport-comparison). Would it be possible to make the docs more generic so that they work in context of other transports as well?